mmpose/models/heads/heatmap_heads/multi_stage_heatmap_head.py

```diff
@@ -0,0 +1,643 @@
+            self.out_channels,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+            norm_cfg=norm_cfg,
+            inplace=False)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward the network. The input heatmaps will be refined.
+
+        Args:
+            x (Tensor): The input heatmaps.
+
+        Returns:
+            Tensor: output heatmaps.
+        """
+        out = self.conv_bn_relu_prm_1(x)
+        out_1 = out
+
+        out_2 = self.global_pooling(out_1)
+        out_2 = out_2.view(out_2.size(0), -1)
+        out_2 = self.middle_path(out_2)
+        out_2 = out_2.unsqueeze(2)
+        out_2 = out_2.unsqueeze(3)
+
+        out_3 = self.bottom_path(out_1)
+        out = out_1 * (1 + out_2 * out_3)
+
+        return out
+
+
+class PredictHeatmap(nn.Module):
+    """Predict the heatmap for an input feature.
+
+    Args:
+        unit_channels (int): Number of input channels.
+        out_channels (int): Number of output channels.
+        out_shape (tuple): Shape of the output heatmaps.
+        use_prm (bool): Whether to use pose refine machine. Default: False.
+        norm_cfg (Config): Config to construct the norm layer.
+            Defaults to ``dict(type='BN')``
+    """
+
+    def __init__(self,
+                 unit_channels: int,
+                 out_channels: int,
+                 out_shape: tuple,
+                 use_prm: bool = False,
+                 norm_cfg: ConfigType = dict(type='BN')):
+
+        super().__init__()
+
+        # Protect mutable default arguments
+        norm_cfg = copy.deepcopy(norm_cfg)
+        self.unit_channels = unit_channels
+        self.out_channels = out_channels
+        self.out_shape = out_shape
+        self.use_prm = use_prm
+        if use_prm:
+            self.prm = PRM(out_channels, norm_cfg=norm_cfg)
+        self.conv_layers = nn.Sequential(
+            ConvModule(
+                unit_channels,
+                unit_channels,
+                kernel_size=1,
+                stride=1,
+                padding=0,
+                norm_cfg=norm_cfg,
+                inplace=False),
+            ConvModule(
+                unit_channels,
+                out_channels,
+                kernel_size=3,
+                stride=1,
+                padding=1,
+                norm_cfg=norm_cfg,
+                act_cfg=None,
+                inplace=False))
+
+    def forward(self, feature: Tensor) -> Tensor:
+        """Forward the network.
+
+        Args:
+            feature (Tensor): The input feature maps.
+
+        Returns:
+            Tensor: output heatmaps.
+        """
+        feature = self.conv_layers(feature)
+        output = nn.functional.interpolate(
+            feature, size=self.out_shape, mode='bilinear', align_corners=True)
+        if self.use_prm:
+            output = self.prm(output)
+        return output
+
+
+@MODELS.register_module()
+class MultiStageMultiUnitHeatmapHead(BaseHead):
```

Comment:
What about placing `MultiStageMultiUnitHeatmapHead` in a separate file, e.g. `msmu_heatmap_head.py`?

Reply:
Good idea. Changed.